### ISSUE TITLE
On-Premise Confluence Server Support

### ISSUE THREAD

**Issue by erauner12** [markdown-confluence/markdown-confluence#91]
> Currently, the plugin only supports Atlassian Cloud instances

Creating a placeholder PR to support On-Premise environments as well

**Comment by andymac4182** [comment]
---

# IF YOU NEED ON-PREMISE SUPPORT PLEASE REACT TO THE FIRST POST SO I CAN UNDERSTAND DEMAND. 

This isn't a small amount of work but I do want to look at it in the future when I have a few days to dig into supporting ADF to Storage format.

---

Thanks for the issue. I don't have a on-premise instance to test against. 

Once I have the next round of changes knocked off I will look at the APIs and what will need to change. 

No guarantees on when I will be able to look at this sorry. Happy for a PR if you would like to contribute 😄

**Comment by erauner12** [comment]
@andymac4182 will see what I can do! linking this here for reference. This raycast tool seems to be able to reach my on-premise instance without issue. Upon setup it asks for either on-premise or cloud and is adjusted accordingly.

https://github.com/raycast/extensions/blob/4f4f5a26a1553aac1549884bc0ad9a41ff9baded/extensions/confluence-search/src/index.tsx

**Comment by erauner12** [comment]
It seems at first glance that the main difference is in the authentication method. On Prem uses a token based authentication instead of basic

```
const headers = {
  Accept: "application/json",
  Authorization:
    prefs.instanceType == "cloud"
      ? "Basic " + Buffer.from(`${prefs.user}:${prefs.token}`).toString("base64")
      : `Bearer ${prefs.token}`,
};
```

ex: `personalAccessToken: this.settings.atlassianApiToken,`

**Comment by erauner12** [comment]
Sorry I have limited experience with Typescript. If you have the time, could you provide a few lines of commands so that I can understand the dev workflow?

mainly, I just want to understand how to make a change to this authentication piece and place built file into: `.obsidian/plugins/confluence-integration/main.js`

**Comment by andymac4182** [comment]
I have added an issue for this https://github.com/obsidian-confluence/obsidian-confluence/issues/162 I will have a look next week 😀

**Comment by lifesign** [comment]
+1 for server support~

**Comment by andymac4182** [comment]
I spun up a Evaluation server today. I use Atlassian Document Format (ADF) to get and send the pages with Confluence. This isn't an option for Data Center. I will have to have a think about how to fit this in and support it. I do a lot of work to ensure I generate the same ADF that Confluence Cloud produces to ensure it round trips exactly the same and produces predictable results. I will probably need to rework a bunch of features to ensure they work across both Cloud and Data Centre consistently or nicely degrade on Data Centre.

I will have a think about this as I work on the code to see how I can make this work.

**Comment by ashi009** [comment]
> I spun up a Evaluation server today. I use Atlassian Document Format (ADF) to get and send the pages with Confluence. This isn't an option for Data Center. I will have to have a think about how to fit this in and support it. I do a lot of work to ensure I generate the same ADF that Confluence Cloud produces to ensure it round trips exactly the same and produces predictable results. I will probably need to rework a bunch of features to ensure they work across both Cloud and Data Centre consistently or nicely degrade on Data Centre.
> 
> I will have a think about this as I work on the code to see how I can make this work.

Use content properties to put a hash marker for the synced markdown file. As the sync is one direction only, so simply compare the marker will be sufficient. If no match, just send everything from top.

P.S. This is my day 2 with the poorly documented confluence REST APIs, so might be wrong.

**Comment by iishabakaev** [comment]
Hi all!
I will be happy to help with add confluence server support.

I play with current code from the master branch and successfully added auth with personal token access and create pages.
But server user fields not compatible with cloud user fields
And content of the page didn't created.

Additionally I found that page don't be overwritten if lastUpdatedBy not equal with the current user.
I think we'll be great to add modal windows with confirmation for overwriting.

**Comment by andymac4182** [comment]
Awesome thanks

Happy to help support this work. 

If you join Discord happy to chat about this as there are others looking for this as well.

**Comment by Fashaun** [comment]
So it could support in latest version?

**Comment by BobGneu** [comment]
Not entirely sure it is portable, but the JIRA issue plugin has a an implementation that appears to be working on JIRA 8 and their [atlassian.net version](https://www.atlassian.com/software/jira/update), which I believe is 9 but I couldn't exactly nail that down. 

https://github.com/marc0l92/obsidian-jira-issue/blob/master/src/settings.ts

This could be a good stepping off point.

**Comment by kosichd** [comment]
Hi all is this a potential? Looking for support for on-prem here.

**Comment by jaredlai-pana** [comment]
Hi, is there an update on this feature yet?

**Comment by nosovk** [comment]
yep, it would be nice to get rid of slow web editor, and stick to obsidian as normal editor

**Comment by Fashaun** [comment]
Which version will it be support?

**Comment by derTobsch** [comment]
Is there any progress on this? Do you need help or is there something preventing you from working on this issue? Let me/us know. Maybe we can support you.

**Comment by titan55441** [comment]
waiting for it！

**Comment by Vlad1mir-D** [comment]
@andymac4182 Hi!
I'm ready to provide on-premise data-center test instances (both latest/LTS) and also sponsor development of this feature.
Is there any way to discuss this with you?
Thanks

**Comment by Vlad1mir-D** [comment]
https://github.com/marcoyohn/markdown-confluence/commit/f121b4945e9d5d6a2d599cc721841780977342b6 resolves author name issue and enables PAT authentication for on-premise but ADF isn't supported by self-hosted version at all so [Confluence storage format](https://confluence.atlassian.com/doc/confluence-storage-format-790796544.html) is the main feature which needs to be implemented.

There is a [tool](https://github.com/highsource/confluence-to-markdown-converter) which implements Confluence storage format to Markdown conversion using XSLT which could be used as a base to implement reverse conversion.

**Comment by DipFlip** [comment]
I made a script to convert my Obsidian markdown notes to an on-premise confluence. I had to use username and password in the API calls instead of API key. See https://github.com/DipFlip/markdown-to-confluence

**Comment by vozer** [comment]
I would really love to have on premise support too (localhost to Confluence server without connections in between bc of vpn restrictions).

Are we any closer with @DipFlip and @Vlad1mir-D s posts?
unfortunately based on the issues resolved in the past year it seems the project is somewhat abandoned. 

Anybody able to do a fork and maintain it?

**Comment by almereyda** [comment]
The DipFlip script worked from here.

**Comment by watarukura** [comment]
There seem to be two issues, one is the conversion to Confluence storage format and the other is accessing with PAT.
Could the first issue be solved by incorporating the following npm package?https://github.com/Shogobg/markdown2confluence
